plugins/net_plugin/net_plugin.cpp

```diff
@@ -2343,11 +2343,14 @@ namespace eosio {
             c->enqueue( note );
          }
          c->peer_syncing_from_us = false;
-         bool on_fork = true;
+         bool on_fork = false;
          try {
             controller& cc = my_impl->chain_plug->chain();
-            on_fork = cc.fork_block_id_for_num( msg.fork_head_num ) != msg.fork_head_id; // thread-safe
-         } catch( ... ) {}
+            std::optional<block_id_type> fork_head_id = cc.fork_block_id_for_num( msg.fork_head_num ); // thread-safe
+            if (fork_head_id) { // possible for LIB to move and fork_head_num not be found if running with no block-log
+               on_fork = fork_head_id != msg.fork_head_id;
+            }
+         } catch( ... ) { on_fork = true; }
          if( on_fork ) {
             request_message req;
```

Comment:
Should we add a warning log here?

Reply:
No this is a micro fork not a fork with diff LIB. I can add a debug level log.